x-pack/plugins/lens/common/expressions/time_scale/time_scale_fn.ts

```diff
@@ -24,31 +24,31 @@ const unitInMs: Record<TimeScaleUnit, number> = {
   d: 1000 * 60 * 60 * 24,
 };
 
-// the datemath plugin always parses dates by using the current default moment time zone.
-// to use the configured time zone, we are temporary switching it just for the calculation.
-
-// The code between this call and the reset in the finally block is not allowed to get async,
-// otherwise the timezone setting can leak out of this function.
-const withChangedTimeZone = <TReturnedValue = unknown>(
-  timeZone: string | undefined,
-  action: () => TReturnedValue
-): TReturnedValue => {
+function safelySetTimeZone(timeZone: string) {
+  const zone = moment.tz.zone(timeZone);
+  if (zone) moment.tz.setDefault(zone.name);
+}
+
+/**
+ * This function can be called both from server side and from the client side. Each of them could have
+ * a different configured timezone. To be sure the time bounds are computed relative to the same passed timezone,
+ * temporarily switch the default moment timezone to the one passed, and switch it back after the calculation is done.
+ */
+export function getTimeBounds(timeRange: TimeRange, timeZone?: string, getForceNow?: () => Date) {
   if (timeZone) {
-    const defaultTimezone = moment().zoneName();
-    try {
-      moment.tz.setDefault(timeZone);
-      return action();
-    } finally {
-      // reset default moment timezone
-      moment.tz.setDefault(defaultTimezone);
-    }
+    // the `defaultZone` property is injected by moment.timezone.
+    // If is not available is it fine to keep undefined because calling setDefault() will automatically reset to default
+    // https://github.com/moment/moment-timezone/blob/2448cdcbe15875bc22ddfbc184794d0a6b568b90/moment-timezone.js#L603
+    // @ts-ignore because is not part of the exposed types unfortunately
```

Comment:
maybe `@ts-expect-error` ?

Reply:
Yea I can change that, but I'm always a bit confused about what `expect error` means: I'm not expecting an error here, the type is just missing, so I'm "ignoring the ts error" because I know something more than ts.
I'm pretty sure this is just because of my English language knowledge.

Reply:
I think it's more to make sure we are alerted if the typing ever gets fixed. `@ts-ignore` can be applied to perfectly valid typescript and the compiler won't complain. `@ts-expect-error` will throw an error if there is no error :D 

Therefore, the danger of `@ts-ignore` is that the original reason it was added could eventually become irrelevant. Then, later a new problem could be introduced into the same ignored code and the compiler wouldn't warn us.

Reply:
And yes, I agree that the wording sounds is a little funky :)

Reply:
Thanks for the explaination, agree,  I've changed this to ts-expect-error here [2adba1f](https://github.com/elastic/kibana/pull/154472/commits/2adba1fbcb9cc8f012035abfc9707d05d226c734)